### ISSUE TITLE
react-jsonschema-form to generate a form itself

### ISSUE THREAD

**Issue by GitToby** [rjsf-team/react-jsonschema-form#1465]
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

Is there a schema that produces a form to generate another form JSON schema? Simmilar to issue #1276, however the owner of that issue vanished.

i.e. a meta  jsonschema form that creates other jsonschema forms 

### Steps to Reproduce
make and run this app:
```javascript
iconst Form = JSONSchemaForm.default;
const schema = {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "$id": "http://json-schema.org/draft-07/schema#",
    "title": "Core schema meta-schema",
...
};
const uiSchema = {
  description: {
    "ui:widget": "textarea"
  }
};

ReactDOM.render(<Form schema={schema} uiSchema={uiSchema}/>, 
             document.getElementById("main"));
```

https://jsfiddle.net/jsToby/u98aygc6/

#### Expected behavior
A form that allows the user to generate a fully valid JSONSchema which produces a form. 

#### Actual behavior

Inputting the definition provided on http://json-schema.org creates the following error: 
```
Core schema meta-schema
Unsupported field schema for field root: Unknown field type object,boolean.
```

### Version

1.8.0

You can usually get this information in your `package.json` or in the file URL if you're using the unpkg one.

**Comment by GitToby** [comment]
@epicfaace it makes sense rjsf cant render a tuple of types as the form should only output one type of result? 

Once I fixed the unsupported schema problem setting `"type": "object"` at the root, i get the following error:
```
Error: Could not find a definition for #.
```

**Comment by Relequestual** [comment]
@GitToby Ah OK.
Yeah, there are a lot of complexities in generating forms for ALL valid schemas.
We would like to t some point work with the teams behind such form generators to define a generalised form vocabulary, post draft-8 release. =]

**Comment by GitToby** [comment]
Yeah, i can take a quick look over the weekend. looks like that PR has gone a bit stale and out of date, i'll look into it anyways!

**Comment by domharrington** [comment]
@GitToby we're using my fork in production at ReadMe to generate dynamic forms from user's OAS files. You can see this in action here: https://preview.readme.io/.

Would love to get it merged in so we don't have to rely on my fork and so others can use it.

**Comment by knilink** [comment]
If the form support the meta schema then it should be possible.
```bash 
node -p "JSON.stringify((new require('ajv')()).getSchema('http://json-schema.org/draft-07/schema').schema,null,2)"
```

**Comment by JulesGosnell** [comment]
Guys,

I'm just starting a project that would really benefit from having this working.

Can anyone give  me the status of this issue ?

We may have some time to throw at it...

Many thanks,

Jules

**Comment by JulesGosnell** [comment]
Looking at the draft-07 meta-schema (https://json-schema.org/draft-07/schema), it looks to me as though it is defined tightly enough to validate a schema, yet not tightly enough to make building one via react-jsonschema-form as easy an experience as it could be...

For example, The draft-07 meta-schema seems to define the set of properties that an object in the schema might have as a big flat space including e.g. "multipleOf" (for use with "number"s)  and "additionaItems" (for use with "array"s) in the same breath. It also includes what appear to be semantically mutually exclusive pairs e.g. "maximum" and "exclusiveMaximum" with no indication of this exclusivity.

I was hoping to use the meta-schema to drive a web-form in which, if I chose to add a new string-property in the resulting schema, I would only be presented with potential attributes for that property which made sense in the context of a string.

As far as I can make out, the introduction of vocabularies in draft-2019-09 does not address this.

In conclusion, it looks to me as though, in order to get the best results out of my meta-schema-generated form, I will have to define my own form-friendly meta-schema ? Has anyone else come to the same conclusion or know of any work in this direction which I might use/extend/contribute-to ?

Many thanks,

Jules

**Comment by epicfaace** [comment]
> I will have to define my own form-friendly meta-schema

Yeah, that's probably correct. Doing this would be a great addition to rjsf, though, as it would clarify what exactly we support and what features of JSON Schema we don't yet support.
